odoo/addons/base/models/ir_module.py

```diff
@@ -642,6 +642,18 @@ def button_upgrade(self):
         self.update_list()
 
         todo = list(self)
+        if 'base' in self.mapped('name'):
+            # If an installed module is only present in the dependency graph through
+            # a new, uninstalled dependency, it will not have been selected yet.
+            # An update of 'base' should also update these modules, and as a consequence,
+            # install the new dependency.
+            todo += list(self.search(
+                [
+                    ('state', '=', 'installed'),
+                    ('name', '!=', 'studio_customization'),
+                    ('id', 'not in', self.ids),
+                ]
```

Comment:
Should this domain be put outside in a separate hook method for others to extend/inject sth when needed?

For example, oca_studio 😄

Reply:
I tested with your code, but I still see the log `odoo.modules.graph: module snailmail: Unmet dependencies: iap_mail`

Reply:
You right, the new dependency is installed regardless. But, how can to avoid the error is not logged?

Reply:
@royleviindoo I don't know, and that is not the aim of this pull request, sorry!